### PR TITLE
Stop reload of live dev preview

### PR DESCRIPTION
**Pull request by ericschv**
Created preference in `LiveDevMultiBrowser.js` designed to disable live dev preview whenever a user imports files. Inside of `src\filesystem\impls\filer\lib\FileImport.js` I disable live dev preview before the import and then undo it after files are loaded.

This fix addresses the bug issue 2446 from the Thimble repo.

Link to the issue: https://github.com/mozilla/thimble.mozilla.org/issues/2446

### CODE REVIEW

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 1]
```diff
@@ -91,8 +92,8 @@ define(function (require, exports, module) {
 
     // XXXBramble: we need to get this loaded so our extension can require/use it later
     require("LiveDevelopment/Servers/BaseServer");
-
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: These types of changes are a bad idea, because they a) aren't related to your actual change; b) make the history of this file (i.e., blame) look like you touched them (when really you didn't).  Probably your editor did this for you?  I'd set your editor to not randomly change whitespace in files for lines you don't touch (those you do touch, this is a good fix).

File: src/extensions/default/bramble/lib/PostMessageTransport.js [hunk 2]
```diff
@@ -223,7 +224,7 @@ define(function (require, exports, module) {
     var _pendingReloadUrl;
 
     // Whether or not to allow reloads in the general case (true by default)
-    var _autoUpdate = true;
+    var _autoUpdate = PreferencesManager.get("livePreviewAutoReload");;
```
Comment (humphd) [inline src/extensions/default/bramble/lib/PostMessageTransport.js]: This won't do what you want.  Here, you're getting the value of the pref when the module is loaded.  But then, during the lifetime of the app, you'll never go back and get it again.  What you should do instead is dump this variable altogether, and instead use `PreferencesManager.get("livePreviewAutoReload")` whenever the other variable was used.  As far as the `setAutoUpdate(value)` function, you can change it to call `PreferencesManager.set("livePreviewAutoReload", value)`.  So basically, we want to move this variable out of this module, and instead have it only live in one place: the preference manager.

File: src/filesystem/impls/filer/lib/FileImport.js [hunk 3]
```diff
@@ -52,10 +53,14 @@ define(function (require, exports, module) {
 
         var strategy = chooseImportStrategy(source);
 
+        PreferencesManager.set("livePreviewAutoReload", false);
```
Comment (humphd) [inline src/filesystem/impls/filer/lib/FileImport.js]: Add a comment above this indicating that we want to pause live preview updates until all files have been imported, due to import order vs dependency order and spurious 404s.

File: src/filesystem/impls/filer/lib/FileImport.js [hunk 4]
```diff
@@ -52,10 +53,14 @@ define(function (require, exports, module) {
...
+
         // If we are given a sub-dir within the project, use that.  Otherwise use project root.
         parentPath = parentPath || BrambleStartupState.project("root");
 
         return strategy.import(source, parentPath, function(err) {
+            PreferencesManager.set("livePreviewAutoReload", true);
```
Comment (humphd) [inline src/filesystem/impls/filer/lib/FileImport.js]: Same thing here, add a comment above this.

File: src/LiveDevelopment/LiveDevMultiBrowser.js (hunk 1 above)
Comment (ericschv) [reply src/LiveDevelopment/LiveDevMultiBrowser.js]: @humphd inside of LiveDevMultiBrowser.js I am seeing these icons in my atom ide. I'm not sure if this is related to the change made at line 94.

livedevmultibrowser_changes

Comment (humphd) [reply src/LiveDevelopment/LiveDevMultiBrowser.js]: If you want me to look at this with you tomorrow, I'd be happy to help you solve it.  It looks like you've got a merge/rebase conflict, which we can sort out and tell git what to take.

Comment (ericschv) [reply src/LiveDevelopment/LiveDevMultiBrowser.js]: That would be great if we could. I appreciate it.

**Comment by humphd** [comment]
@ericschv I think you know this, but in case not, you're failing lint:

```
/home/travis/build/mozilla/brackets/src/LiveDevelopment/LiveDevMultiBrowser.js
  978:40  error  'S' is not defined  no-undef
  978:41  error  Missing semicolon   semi
✖ 2 problems (2 errors, 0 warnings)
```

Can you fix those two issues and push a new commit?

**Comment by ericschv** [comment]
@humphd yes I am aware. I will make the commit now.

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 5]
```diff
@@ -958,6 +959,8 @@ define(function (require, exports, module) {
 
     EventDispatcher.makeEventDispatcher(exports);
 
+    // XXXBramble: we need livePreviewAutoReload preference available to both core and extension
+
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: It looks like your preference definition got lost somehow.  Can you add it back?

File: src/extensions/default/bramble/lib/PostMessageTransport.js [hunk 6]
```diff
@@ -222,15 +223,13 @@ define(function (require, exports, module) {
...
-
+    // Whether or not to allow reloads in the general case
     function setAutoUpdate(value) {
-        _autoUpdate = value;
+        PreferencesManager.set("livePreviewAutoReload",value);
 
         // Force a reload if we switch back to auto-updates
-        if(_autoUpdate) {
-            reload(true);
+        if(PreferencesManager.get("livePreviewAutoReload")) {
```
Comment (humphd) [inline src/extensions/default/bramble/lib/PostMessageTransport.js]: You can undo this change, and make it:

```js
if(value) {
    reload(true);
}
```

File: src/LiveDevelopment/LiveDevMultiBrowser.js (hunk 5 above)
Comment (ericschv) [reply src/LiveDevelopment/LiveDevMultiBrowser.js]: My apologies . Must have missed that when I did `checkout` from before.

**Comment by ericschv** [comment]
@humphd I made the changes you requested

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 7]
```diff
@@ -978,4 +982,4 @@ define(function (require, exports, module) {
     exports.getCurrentProjectServerConfig = getCurrentProjectServerConfig;
     exports.setTransport        = setTransport;
     exports.setLauncher         = setLauncher;
-});
+});
```
Comment (gideonthomas) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: undo this change

**Comment by gideonthomas** [comment]
Thanks @ericschv, nice work!
